File: src/aiBridgeStatus.js
function envFlag(name, env = process.env) {
  const raw = env[name];
  if (!raw) return false;
  return ["1", "true", "yes", "on"].includes(raw.trim().toLowerCase());
}

export function hostFromUrl(rawUrl) {
  const trimmed = typeof rawUrl === "string" ? rawUrl.trim() : "";
  if (!trimmed) {
    return null;
  }
  try {
    return new URL(trimmed).host;
  } catch {
    return "invalid-url";
  }
}

/**
 * Non-secret AI bridge snapshot for /health and startup logs.
 */
export function buildAiBridgeStatus(env = process.env) {
  const baseUrl = env.AI_ORCHESTRATION_BASE_URL?.trim() || "";
  const suggestFlag = envFlag("AI_SUGGEST_VENDORS_ENABLED", env);
  const instructionFlag = envFlag("AI_INSTRUCTION_PACK_ENABLED", env);
  const baseUrlSet = Boolean(baseUrl);

  return {
    orchestration_base_url_set: baseUrlSet,
    orchestration_host: hostFromUrl(baseUrl),
    suggest_vendors_flag: suggestFlag,
    instruction_pack_flag: instructionFlag,
    suggest_vendors_path_active: baseUrlSet && suggestFlag,
    instruction_pack_path_active: baseUrlSet && instructionFlag,
    internal_api_key_set: Boolean(env.AI_ORCHESTRATION_INTERNAL_API_KEY?.trim()),
    orchestration_timeout_ms: Number(env.AI_ORCHESTRATION_TIMEOUT_MS || 15000),
    instruction_pack_timeout_ms: Number(
      env.AI_ORCHESTRATION_INSTRUCTION_PACK_TIMEOUT_MS ||
        env.AI_ORCHESTRATION_TIMEOUT_MS ||
        60000
    )
  };
}

export function buildStartupConfig(env = process.env) {
  return {
    service: "integration-service",
    user_service_host: hostFromUrl(env.USER_SERVICE_BASE_URL),
    database_url_set: Boolean(env.DATABASE_URL?.trim()),
    web_cors_origins_set: Boolean(env.WEB_CORS_ORIGINS?.trim()),
    ai: buildAiBridgeStatus(env)
  };
}

export function explainMockSuggestVendorsReason(env = process.env) {
  const baseUrl = env.AI_ORCHESTRATION_BASE_URL?.trim() || "";
  if (!baseUrl) {
    return "AI_ORCHESTRATION_BASE_URL is unset";
  }
  if (!envFlag("AI_SUGGEST_VENDORS_ENABLED", env)) {
    return "AI_SUGGEST_VENDORS_ENABLED is not true";
  }
  return "AI orchestration client not configured";
}

export function explainInstructionPackMockReason(env = process.env) {
  const baseUrl = env.AI_ORCHESTRATION_BASE_URL?.trim() || "";
  if (!baseUrl) {
    return "AI_ORCHESTRATION_BASE_URL is unset";
  }
  if (!envFlag("AI_INSTRUCTION_PACK_ENABLED", env)) {
    return "AI_INSTRUCTION_PACK_ENABLED is not true";
  }
  return "AI orchestration client not configured";
}

File: src/main/java/org/sharingbridge/integration/service/AiBridgeStatus.java
package org.sharingbridge.integration.service;

import java.net.URI;
import java.util.LinkedHashMap;
import java.util.Locale;
import java.util.Map;

/**
 * Non-secret AI bridge snapshot for {@code /health}, matching Node {@code aiBridgeStatus.js}.
 */
public final class AiBridgeStatus {

    private AiBridgeStatus() {}

    public static Map<String, Object> fromEnvironment() {
        return build(System.getenv());
    }

    public static Map<String, Object> build(Map<String, String> env) {
        String baseUrl = trim(env.get("AI_ORCHESTRATION_BASE_URL"));
        boolean suggestFlag = envFlag(env.get("AI_SUGGEST_VENDORS_ENABLED"));
        boolean instructionFlag = envFlag(env.get("AI_INSTRUCTION_PACK_ENABLED"));
        boolean baseUrlSet = !baseUrl.isEmpty();

        Map<String, Object> status = new LinkedHashMap<>();
        status.put("orchestration_base_url_set", baseUrlSet);
        status.put("orchestration_host", hostFromUrl(baseUrl));
        status.put("suggest_vendors_flag", suggestFlag);
        status.put("instruction_pack_flag", instructionFlag);
        status.put("suggest_vendors_path_active", baseUrlSet && suggestFlag);
        status.put("instruction_pack_path_active", baseUrlSet && instructionFlag);
        status.put(
                "internal_api_key_set",
                !trim(env.get("AI_ORCHESTRATION_INTERNAL_API_KEY")).isEmpty());
        status.put(
                "suggest_vendors_timeout_ms",
                readNumber(env.get("AI_ORCHESTRATION_SUGGEST_VENDORS_TIMEOUT_MS"), 15_000));
        status.put(
                "instruction_pack_timeout_ms",
                readNumber(env.get("AI_ORCHESTRATION_INSTRUCTION_PACK_TIMEOUT_MS"), 60_000));
        return status;
    }

    public static String hostFromUrl(String rawUrl) {
        String trimmed = trim(rawUrl);
        if (trimmed.isEmpty()) {
            return null;
        }
        try {
            URI uri = URI.create(trimmed);
            String host = uri.getHost();
            if (host == null || host.isBlank()) {
                int port = uri.getPort();
                if (port > 0 && uri.getAuthority() != null) {
                    return uri.getAuthority();
                }
                return "invalid-url";
            }
            int port = uri.getPort();
            return port > 0 ? host + ":" + port : host;
        } catch (IllegalArgumentException ex) {
            return "invalid-url";
        }
    }

    private static boolean envFlag(String raw) {
        if (raw == null || raw.isBlank()) {
            return false;
        }
        return switch (raw.trim().toLowerCase(Locale.ROOT)) {
            case "1", "true", "yes", "on" -> true;
            default -> false;
        };
    }

    private static double readNumber(String raw, double fallback) {
        if (raw == null || raw.isBlank()) {
            return fallback;
        }
        try {
            return Double.parseDouble(raw.trim());
        } catch (NumberFormatException ex) {
            return Double.NaN;
        }
    }

    private static String trim(String value) {
        return value == null ? "" : value.trim();
    }
}
